diff -up gvfs-1.4.3/gconf/Makefile.am.orig gvfs-1.4.3/gconf/Makefile.am
--- gvfs-1.4.3/gconf/Makefile.am.orig	2014-07-11 11:42:49.064943260 +0200
+++ gvfs-1.4.3/gconf/Makefile.am	2014-07-11 11:43:13.956004037 +0200
@@ -19,7 +19,6 @@ libgiogconf_la_CFLAGS =		\
 	$(GCONF_CFLAGS)                         \
 	-DGIO_MODULE_DIR=\"$(GIO_MODULE_DIR)\"	\
 	-DGVFS_LOCALEDIR=\""$(localedir)"\"	\
-	-DG_DISABLE_DEPRECATED			\
 	$(NULL)
 
 libgiogconf_la_LDFLAGS = 	\
diff -up gvfs-1.4.3/gconf/Makefile.in.orig gvfs-1.4.3/gconf/Makefile.in
--- gvfs-1.4.3/gconf/Makefile.in.orig	2014-07-11 11:42:40.711920833 +0200
+++ gvfs-1.4.3/gconf/Makefile.in	2014-07-11 11:43:04.585983256 +0200
@@ -292,7 +292,6 @@ libgiogconf_la_CFLAGS = \
 	$(GCONF_CFLAGS)                         \
 	-DGIO_MODULE_DIR=\"$(GIO_MODULE_DIR)\"	\
 	-DGVFS_LOCALEDIR=\""$(localedir)"\"	\
-	-DG_DISABLE_DEPRECATED			\
 	$(NULL)
 
 libgiogconf_la_LDFLAGS = \
