diff --git a/gconf/gapplookupgconf.c b/gconf/gapplookupgconf.c
index 5d477ba..16a939f 100644
--- a/gconf/gapplookupgconf.c
+++ b/gconf/gapplookupgconf.c
@@ -108,7 +108,8 @@ get_default_for_uri_scheme (GDesktopAppInfoLookup *lookup,
 {
   GAppInfo *appinfo;
   GConfClient *client;
-  char *command_key, *enabled_key, *terminal_key, *command;
+  char *command_key, *enabled_key, *terminal_key;
+  char *command = NULL;
   gboolean enabled, needs_terminal;
   GAppInfoCreateFlags flags;
 
@@ -120,7 +121,9 @@ get_default_for_uri_scheme (GDesktopAppInfoLookup *lookup,
                              uri_scheme,
                              "/command",
                              NULL);
-  command = gconf_client_get_string (client,
+
+  if(gconf_valid_key(command_key,NULL))
+    command = gconf_client_get_string (client,
                                      command_key,
                                      NULL);
   g_free (command_key);
