diff -up gvfs-1.4.3/metadata/metatree.c.orig gvfs-1.4.3/metadata/metatree.c
--- gvfs-1.4.3/metadata/metatree.c.orig	2013-11-14 14:36:16.513801730 +0100
+++ gvfs-1.4.3/metadata/metatree.c	2013-11-14 14:49:49.240261772 +0100
@@ -1209,7 +1209,7 @@ meta_journal_iterate (MetaJournal *journ
 		      gpointer user_data)
 {
   MetaJournalEntry *entry;
-  guint32 *sizep;
+  guint32 *sizep, size;
   char *journal_path, *journal_key, *source_path;
   char *path_copy, *value;
   gboolean res;
@@ -1224,8 +1224,11 @@ meta_journal_iterate (MetaJournal *journ
   while (entry > journal->first_entry)
     {
       sizep = (guint32 *)entry;
-      entry = (MetaJournalEntry *)((char *)entry - GUINT32_FROM_BE (*(sizep-1)));
-      if (GUINT32_FROM_BE (*(sizep)) < sizeof (MetaJournalEntry) && entry > journal->first_entry)
+      size = GUINT32_FROM_BE (*(sizep-1));
+      entry = (MetaJournalEntry *)((char *)entry - size);
+      if (size < sizeof (MetaJournalEntry) ||
+	  entry < journal->first_entry ||
+	  entry >= journal->last_entry)
         {
           g_warning ("meta_journal_iterate: found short sized entry, possible journal corruption\n");
           break;
@@ -2269,11 +2272,13 @@ apply_journal_to_builder (MetaTree *tree
 
       sizep = (guint32 *)entry;
       entry = (MetaJournalEntry *)((char *)entry + GUINT32_FROM_BE (*(sizep)));
-      if (GUINT32_FROM_BE (*(sizep)) < sizeof (MetaJournalEntry) && entry < journal->last_entry)
+      if (GUINT32_FROM_BE (*(sizep)) < sizeof (MetaJournalEntry) ||
+	  entry < journal->first_entry ||
+	  entry > journal->last_entry)
         {
           /* This shouldn't happen, we found an entry that is shorter than its data */
           /* See https://bugzilla.gnome.org/show_bug.cgi?id=637095 for discussion */
-          g_warning ("apply_journal_to_builder: found short sized entry, possible journal corruption\n");
+          g_warning ("apply_journal_to_builder: found wrong sized entry, possible journal corruption\n");
           break;
         }
     }
