--- gvfs-1.4.3/monitor/gdu/ggduvolumemonitor.c.handle-null-gdu-pool	2014-12-17 14:50:56.807156067 +0100
+++ gvfs-1.4.3/monitor/gdu/ggduvolumemonitor.c	2014-12-17 15:36:59.033078261 +0100
@@ -432,25 +432,28 @@
 
   monitor->pool = gdu_pool_new ();
 
-  g_signal_connect (monitor->pool,
-                    "presentable_added",
-                    G_CALLBACK (presentable_added),
-                    monitor);
-
-  g_signal_connect (monitor->pool,
-                    "presentable_removed",
-                    G_CALLBACK (presentable_removed),
-                    monitor);
-
-  g_signal_connect (monitor->pool,
-                    "presentable_changed",
-                    G_CALLBACK (presentable_changed),
-                    monitor);
-
-  g_signal_connect (monitor->pool,
-                    "presentable_job_changed",
-                    G_CALLBACK (presentable_job_changed),
-                    monitor);
+  if (monitor->pool != NULL)
+    {
+      g_signal_connect (monitor->pool,
+                        "presentable_added",
+                        G_CALLBACK (presentable_added),
+                        monitor);
+
+      g_signal_connect (monitor->pool,
+                        "presentable_removed",
+                        G_CALLBACK (presentable_removed),
+                        monitor);
+
+      g_signal_connect (monitor->pool,
+                        "presentable_changed",
+                        G_CALLBACK (presentable_changed),
+                        monitor);
+
+      g_signal_connect (monitor->pool,
+                        "presentable_job_changed",
+                        G_CALLBACK (presentable_job_changed),
+                        monitor);
+    }
 
   update_all (monitor, FALSE);
 
@@ -1217,6 +1220,9 @@
   GGduDrive *drive;
   GList *fstab_mount_points;
 
+  if (monitor->pool == NULL)
+    return;
+
   fstab_mount_points = g_unix_mount_points_get (NULL);
 
   cur_drives = NULL;
@@ -1300,6 +1306,9 @@
   GGduDrive *drive;
   GList *fstab_mount_points;
 
+  if (monitor->pool == NULL)
+    return;
+
   fstab_mount_points = g_unix_mount_points_get (NULL);
 
   cur_volumes = NULL;
@@ -1428,14 +1437,15 @@
       if (g_str_has_prefix (device_file, "/dev/"))
         {
           gchar resolved_path[PATH_MAX];
-          GduDevice *device;
+          GduDevice *device = NULL;
 
           /* doesn't exist */
           if (realpath (device_file, resolved_path) != 0)
             continue;
 
           /* is handled by DKD */
-          device = gdu_pool_get_by_device_file (monitor->pool, resolved_path);
+          if (monitor->pool != NULL)
+            device = gdu_pool_get_by_device_file (monitor->pool, resolved_path);
           if (device != NULL)
             {
               g_object_unref (device);
@@ -1650,6 +1660,9 @@
    *
    */
 
+  if (monitor->pool == NULL)
+    return;
+
   cur_discs = NULL;
   for (l = monitor->disc_volumes; l != NULL; l = l->next)
     cur_discs = g_list_prepend (cur_discs, g_gdu_volume_get_presentable (G_GDU_VOLUME (l->data)));
