From 5ce0a03107246b6fce7314ad2bac01c426ca85c1 Mon Sep 17 00:00:00 2001
From: Jonh Wendell <jwendell@gnome.org>
Date: Tue, 25 Aug 2009 17:44:47 +0000
Subject: Fixed description for gvfs-mkdir

https://bugzilla.gnome.org/show_bug.cgi?id=593045
---
diff --git a/programs/gvfs-mkdir.c b/programs/gvfs-mkdir.c
index a031dab..0359172 100644
--- a/programs/gvfs-mkdir.c
+++ b/programs/gvfs-mkdir.c
@@ -47,7 +47,7 @@ main (int argc, char *argv[])
   g_type_init ();
 
   error = NULL;
-  context = g_option_context_new (_("- delete files"));
+  context = g_option_context_new (_("- create directories"));
   g_option_context_add_main_entries (context, entries, GETTEXT_PACKAGE);
   g_option_context_parse (context, &argc, &argv, &error);
   g_option_context_free (context);
--
cgit v0.8.3.1
