From 070b4c4968591b612f3db1aafd9bba07a4ee0004 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Fri, 12 Nov 2010 15:37:30 +0000
Subject: sftp: Close pty master in child process

Prevent fd leakage and possible selinux alert.
---
diff --git a/daemon/pty_open.c b/daemon/pty_open.c
index 6f6d83e..22fa6b9 100644
--- a/daemon/pty_open.c
+++ b/daemon/pty_open.c
@@ -448,6 +448,7 @@ _pty_fork_on_pty_name(const char *path, int parent_fd, char **env_add,
 		break;
 	case 0:
 		/* Child. Close the parent's ends of the pipes. */
+		close(parent_fd);
 		close(ready_a[0]);
 		close(ready_b[1]);
 	
--
cgit v0.8.3.1
