diff -up gvfs-1.4.3/client/gdaemonfile.c.orig gvfs-1.4.3/client/gdaemonfile.c
--- gvfs-1.4.3/client/gdaemonfile.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/client/gdaemonfile.c	2013-11-14 14:26:15.789739885 +0100
@@ -37,7 +37,7 @@
 #include <gdaemonfilemonitor.h>
 #include <gdaemonfileenumerator.h>
 #include <glib/gi18n-lib.h>
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gmountoperationdbus.h"
 #include <gio/gio.h>
 #include "metatree.h"
diff -up gvfs-1.4.3/client/gdaemonfilemonitor.c.orig gvfs-1.4.3/client/gdaemonfilemonitor.c
--- gvfs-1.4.3/client/gdaemonfilemonitor.c.orig	2009-04-20 12:41:37.000000000 +0200
+++ gvfs-1.4.3/client/gdaemonfilemonitor.c	2013-11-14 14:26:15.796739892 +0100
@@ -27,7 +27,7 @@
 #include <gio/gio.h>
 #include <gvfsdaemondbus.h>
 #include <gvfsdaemonprotocol.h>
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gmountspec.h"
 #include "gdaemonfile.h"
 
diff -up gvfs-1.4.3/client/gdaemonmount.c.orig gvfs-1.4.3/client/gdaemonmount.c
--- gvfs-1.4.3/client/gdaemonmount.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/client/gdaemonmount.c	2013-11-14 14:26:15.803739905 +0100
@@ -32,7 +32,7 @@
 #include "gvfsdaemondbus.h"
 #include "gdaemonfile.h"
 #include "gvfsdaemonprotocol.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gmountsource.h"
 #include "gmountoperationdbus.h"
 
diff -up gvfs-1.4.3/client/gdaemonvfs.c.orig gvfs-1.4.3/client/gdaemonvfs.c
--- gvfs-1.4.3/client/gdaemonvfs.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/client/gdaemonvfs.c	2013-11-14 14:26:15.806739911 +0100
@@ -33,7 +33,7 @@
 #include <gvfsdaemonprotocol.h>
 #include <gmodule.h>
 #include "gvfsdaemondbus.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gmountspec.h"
 #include "gvfsurimapper.h"
 #include "gdaemonvolumemonitor.h"
diff -up gvfs-1.4.3/client/gvfsdaemondbus.c.orig gvfs-1.4.3/client/gvfsdaemondbus.c
--- gvfs-1.4.3/client/gvfsdaemondbus.c.orig	2009-08-21 20:26:35.000000000 +0200
+++ gvfs-1.4.3/client/gvfsdaemondbus.c	2013-11-14 14:26:15.807739913 +0100
@@ -36,7 +36,7 @@
 #include "gvfsdaemondbus.h"
 #include <gvfsdaemonprotocol.h>
 #include <gdaemonvfs.h>
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gsysutils.h"
 
 /* Extra vfs-specific data for DBusConnections */
diff -up gvfs-1.4.3/client/gvfsfusedaemon.c.orig gvfs-1.4.3/client/gvfsfusedaemon.c
--- gvfs-1.4.3/client/gvfsfusedaemon.c.orig	2009-11-27 15:15:19.000000000 +0100
+++ gvfs-1.4.3/client/gvfsfusedaemon.c	2013-11-14 14:26:15.813739925 +0100
@@ -46,7 +46,7 @@
 /* stuff from common/ */
 #include <gdaemonmount.h>
 #include <gvfsdaemonprotocol.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #define FUSE_USE_VERSION 26
 #include <fuse.h>
diff -up gvfs-1.4.3/client/gvfsiconloadable.c.orig gvfs-1.4.3/client/gvfsiconloadable.c
--- gvfs-1.4.3/client/gvfsiconloadable.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/client/gvfsiconloadable.c	2013-11-14 14:26:15.824739946 +0100
@@ -31,7 +31,7 @@
 #include "gdaemonvfs.h"
 #include "gdaemonfileinputstream.h"
 #include "gvfsdaemonprotocol.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 
 /* see comment in common/giconvfs.c for why the GLoadableIcon interface is here */
 
diff -up gvfs-1.4.3/common/gdbusutils.c.orig gvfs-1.4.3/common/gdbusutils.c
--- gvfs-1.4.3/common/gdbusutils.c.orig	2009-05-23 20:08:21.000000000 +0200
+++ gvfs-1.4.3/common/gdbusutils.c	2013-11-14 15:38:23.985957710 +0100
@@ -26,7 +26,7 @@
 #include <string.h>
 
 #include <glib/gi18n-lib.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 #include <gio/gio.h>
 
 void
diff -up gvfs-1.4.3/common/gdbusutils.h.orig gvfs-1.4.3/common/gdbusutils.h
--- gvfs-1.4.3/common/gdbusutils.h.orig	2009-04-20 12:41:37.000000000 +0200
+++ gvfs-1.4.3/common/gdbusutils.h	2013-11-14 15:38:31.769973257 +0100
@@ -20,8 +20,8 @@
  * Author: Alexander Larsson <alexl@redhat.com>
  */
 
-#ifndef __G_DBUS_UTILS_H__
-#define __G_DBUS_UTILS_H__
+#ifndef __GVFS_DBUS_UTILS_H__
+#define __GVFS_DBUS_UTILS_H__
 
 #include <glib.h>
 #include <dbus/dbus.h>
@@ -92,4 +92,4 @@ void        _g_dbus_connection_call_asyn
 G_END_DECLS
 
 
-#endif /* __G_DBUS_UTILS_H__ */
+#endif /* __GVFS_DBUS_UTILS_H__ */
diff -up gvfs-1.4.3/common/gmountoperationdbus.c.orig gvfs-1.4.3/common/gmountoperationdbus.c
--- gvfs-1.4.3/common/gmountoperationdbus.c.orig	2009-07-14 13:07:59.000000000 +0200
+++ gvfs-1.4.3/common/gmountoperationdbus.c	2013-11-14 14:26:15.847739992 +0100
@@ -31,7 +31,7 @@
 #include <gio/gio.h>
 #include "gmountoperationdbus.h"
 #include "gvfsdaemonprotocol.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include <glib/gi18n-lib.h>
 
 typedef struct 
diff -up gvfs-1.4.3/common/gmountsource.c.orig gvfs-1.4.3/common/gmountsource.c
--- gvfs-1.4.3/common/gmountsource.c.orig	2009-07-14 13:07:59.000000000 +0200
+++ gvfs-1.4.3/common/gmountsource.c	2013-11-14 14:26:15.851739999 +0100
@@ -23,7 +23,7 @@
 #include <config.h>
 
 #include <gmountsource.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 #include <gio/gio.h>
 #include <gvfsdaemonprotocol.h>
 
diff -up gvfs-1.4.3/common/gmountspec.c.orig gvfs-1.4.3/common/gmountspec.c
--- gvfs-1.4.3/common/gmountspec.c.orig	2009-11-19 11:43:19.000000000 +0100
+++ gvfs-1.4.3/common/gmountspec.c	2013-11-14 14:26:15.852740001 +0100
@@ -27,7 +27,7 @@
 
 #include <glib/gi18n-lib.h>
 
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gmountspec.h"
 
 static GHashTable *unique_hash = NULL;
diff -up gvfs-1.4.3/common/gmounttracker.c.orig gvfs-1.4.3/common/gmounttracker.c
--- gvfs-1.4.3/common/gmounttracker.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/common/gmounttracker.c	2013-11-14 14:26:15.866740029 +0100
@@ -25,7 +25,7 @@
 #include <string.h>
 
 #include <gmounttracker.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 #include <gvfsdaemonprotocol.h>
 
 enum {
diff -up gvfs-1.4.3/common/gvfsdaemonprotocol.c.orig gvfs-1.4.3/common/gvfsdaemonprotocol.c
--- gvfs-1.4.3/common/gvfsdaemonprotocol.c.orig	2013-11-14 14:25:35.485660558 +0100
+++ gvfs-1.4.3/common/gvfsdaemonprotocol.c	2013-11-14 14:26:15.866740029 +0100
@@ -26,7 +26,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n-lib.h>
 #include <gvfsdaemonprotocol.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 #include <gio/gio.h>
 
 static const char *
diff -up gvfs-1.4.3/common/Makefile.am.orig gvfs-1.4.3/common/Makefile.am
--- gvfs-1.4.3/common/Makefile.am.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/common/Makefile.am	2013-11-14 14:26:15.834739966 +0100
@@ -10,7 +10,7 @@ INCLUDES = -I$(top_srcdir) -I$(top_srcdi
 
 libgvfscommon_la_SOURCES = 	\
 	gsysutils.c gsysutils.h \
-	gdbusutils.c gdbusutils.h \
+	gvfsdbusutils.c gvfsdbusutils.h \
 	gmountspec.c gmountspec.h \
 	gmountoperationdbus.c gmountoperationdbus.h \
 	gmountsource.c gmountsource.h \
diff -up gvfs-1.4.3/daemon/gvfsbackend.c.orig gvfs-1.4.3/daemon/gvfsbackend.c
--- gvfs-1.4.3/daemon/gvfsbackend.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/daemon/gvfsbackend.c	2013-11-14 14:26:15.876740049 +0100
@@ -58,7 +58,7 @@
 #include <gvfsjobpull.h>
 #include <gvfsjobsetattribute.h>
 #include <gvfsjobqueryattributes.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 enum {
   PROP_0,
diff -up gvfs-1.4.3/daemon/gvfsbackend.h.orig gvfs-1.4.3/daemon/gvfsbackend.h
--- gvfs-1.4.3/daemon/gvfsbackend.h.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/daemon/gvfsbackend.h	2013-11-14 14:26:15.884740065 +0100
@@ -28,7 +28,7 @@
 #include <gvfsdaemon.h>
 #include <gvfsjob.h>
 #include <gmountspec.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 G_BEGIN_DECLS
 
diff -up gvfs-1.4.3/daemon/gvfsdaemon.c.orig gvfs-1.4.3/daemon/gvfsdaemon.c
--- gvfs-1.4.3/daemon/gvfsdaemon.c.orig	2013-11-14 14:25:35.465660519 +0100
+++ gvfs-1.4.3/daemon/gvfsdaemon.c	2013-11-14 14:26:15.885740067 +0100
@@ -42,7 +42,7 @@
 #include <gvfsjobmount.h>
 #include <gvfsjobopenforread.h>
 #include <gvfsjobopenforwrite.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 enum {
   PROP_0
diff -up gvfs-1.4.3/daemon/gvfsdaemonutils.c.orig gvfs-1.4.3/daemon/gvfsdaemonutils.c
--- gvfs-1.4.3/daemon/gvfsdaemonutils.c.orig	2009-04-20 12:41:37.000000000 +0200
+++ gvfs-1.4.3/daemon/gvfsdaemonutils.c	2013-11-14 14:26:15.885740067 +0100
@@ -35,7 +35,7 @@
 #include <glib/gi18n.h>
 
 #include <gio/gio.h>
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gsysutils.h"
 #include "gvfsdaemonutils.h"
 #include "gvfsdaemonprotocol.h"
diff -up gvfs-1.4.3/daemon/gvfsjobcopy.c.orig gvfs-1.4.3/daemon/gvfsjobcopy.c
--- gvfs-1.4.3/daemon/gvfsjobcopy.c.orig	2013-11-14 14:25:35.558660702 +0100
+++ gvfs-1.4.3/daemon/gvfsjobcopy.c	2013-11-14 14:26:15.885740067 +0100
@@ -32,7 +32,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobcopy.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobCopy, g_vfs_job_copy, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobcreatemonitor.c.orig gvfs-1.4.3/daemon/gvfsjobcreatemonitor.c
--- gvfs-1.4.3/daemon/gvfsjobcreatemonitor.c.orig	2013-11-14 14:25:35.558660702 +0100
+++ gvfs-1.4.3/daemon/gvfsjobcreatemonitor.c	2013-11-14 14:26:15.886740069 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobcreatemonitor.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobCreateMonitor, g_vfs_job_create_monitor, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobdbus.c.orig gvfs-1.4.3/daemon/gvfsjobdbus.c
--- gvfs-1.4.3/daemon/gvfsjobdbus.c.orig	2009-04-20 12:41:37.000000000 +0200
+++ gvfs-1.4.3/daemon/gvfsjobdbus.c	2013-11-14 14:26:15.886740069 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobdbus.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 
 G_DEFINE_TYPE (GVfsJobDBus, g_vfs_job_dbus, G_VFS_TYPE_JOB)
 
diff -up gvfs-1.4.3/daemon/gvfsjobdelete.c.orig gvfs-1.4.3/daemon/gvfsjobdelete.c
--- gvfs-1.4.3/daemon/gvfsjobdelete.c.orig	2013-11-14 14:25:35.558660702 +0100
+++ gvfs-1.4.3/daemon/gvfsjobdelete.c	2013-11-14 14:26:15.886740069 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobdelete.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobDelete, g_vfs_job_delete, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobenumerate.c.orig gvfs-1.4.3/daemon/gvfsjobenumerate.c
--- gvfs-1.4.3/daemon/gvfsjobenumerate.c.orig	2013-11-14 14:25:35.558660702 +0100
+++ gvfs-1.4.3/daemon/gvfsjobenumerate.c	2013-11-14 14:26:15.886740069 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobenumerate.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobEnumerate, g_vfs_job_enumerate, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobmakedirectory.c.orig gvfs-1.4.3/daemon/gvfsjobmakedirectory.c
--- gvfs-1.4.3/daemon/gvfsjobmakedirectory.c.orig	2013-11-14 14:25:35.558660702 +0100
+++ gvfs-1.4.3/daemon/gvfsjobmakedirectory.c	2013-11-14 14:26:15.887740070 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobmakedirectory.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobMakeDirectory, g_vfs_job_make_directory, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobmakesymlink.c.orig gvfs-1.4.3/daemon/gvfsjobmakesymlink.c
--- gvfs-1.4.3/daemon/gvfsjobmakesymlink.c.orig	2013-11-14 14:25:35.558660702 +0100
+++ gvfs-1.4.3/daemon/gvfsjobmakesymlink.c	2013-11-14 14:26:15.887740070 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobmakesymlink.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobMakeSymlink, g_vfs_job_make_symlink, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobmount.c.orig gvfs-1.4.3/daemon/gvfsjobmount.c
--- gvfs-1.4.3/daemon/gvfsjobmount.c.orig	2009-04-20 12:41:37.000000000 +0200
+++ gvfs-1.4.3/daemon/gvfsjobmount.c	2013-11-14 14:26:15.891740078 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobmount.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobMount, g_vfs_job_mount, G_VFS_TYPE_JOB)
diff -up gvfs-1.4.3/daemon/gvfsjobmountmountable.c.orig gvfs-1.4.3/daemon/gvfsjobmountmountable.c
--- gvfs-1.4.3/daemon/gvfsjobmountmountable.c.orig	2013-11-14 14:25:35.558660702 +0100
+++ gvfs-1.4.3/daemon/gvfsjobmountmountable.c	2013-11-14 14:26:15.891740078 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobmountmountable.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobMountMountable, g_vfs_job_mount_mountable, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobmove.c.orig gvfs-1.4.3/daemon/gvfsjobmove.c
--- gvfs-1.4.3/daemon/gvfsjobmove.c.orig	2013-11-14 14:25:35.559660703 +0100
+++ gvfs-1.4.3/daemon/gvfsjobmove.c	2013-11-14 14:26:15.891740078 +0100
@@ -32,7 +32,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobmove.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobMove, g_vfs_job_move, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobopenforread.c.orig gvfs-1.4.3/daemon/gvfsjobopenforread.c
--- gvfs-1.4.3/daemon/gvfsjobopenforread.c.orig	2013-11-14 14:25:35.559660703 +0100
+++ gvfs-1.4.3/daemon/gvfsjobopenforread.c	2013-11-14 14:26:15.891740078 +0100
@@ -32,7 +32,7 @@
 #include <glib/gi18n.h>
 #include "gvfsreadchannel.h"
 #include "gvfsjobopenforread.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobOpenForRead, g_vfs_job_open_for_read, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobopenforwrite.c.orig gvfs-1.4.3/daemon/gvfsjobopenforwrite.c
--- gvfs-1.4.3/daemon/gvfsjobopenforwrite.c.orig	2013-11-14 14:25:35.559660703 +0100
+++ gvfs-1.4.3/daemon/gvfsjobopenforwrite.c	2013-11-14 14:26:15.892740081 +0100
@@ -32,7 +32,7 @@
 #include <glib/gi18n.h>
 #include "gvfswritechannel.h"
 #include "gvfsjobopenforwrite.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobOpenForWrite, g_vfs_job_open_for_write, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobopeniconforread.c.orig gvfs-1.4.3/daemon/gvfsjobopeniconforread.c
--- gvfs-1.4.3/daemon/gvfsjobopeniconforread.c.orig	2013-11-14 14:25:35.559660703 +0100
+++ gvfs-1.4.3/daemon/gvfsjobopeniconforread.c	2013-11-14 14:26:15.892740081 +0100
@@ -32,7 +32,7 @@
 #include <glib/gi18n.h>
 #include "gvfsreadchannel.h"
 #include "gvfsjobopeniconforread.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobOpenIconForRead, g_vfs_job_open_icon_for_read, G_VFS_TYPE_JOB_OPEN_FOR_READ)
diff -up gvfs-1.4.3/daemon/gvfsjobpollmountable.c.orig gvfs-1.4.3/daemon/gvfsjobpollmountable.c
--- gvfs-1.4.3/daemon/gvfsjobpollmountable.c.orig	2013-11-14 14:25:35.559660703 +0100
+++ gvfs-1.4.3/daemon/gvfsjobpollmountable.c	2013-11-14 14:26:15.892740081 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobpollmountable.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobPollMountable, g_vfs_job_poll_mountable, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobpull.c.orig gvfs-1.4.3/daemon/gvfsjobpull.c
--- gvfs-1.4.3/daemon/gvfsjobpull.c.orig	2013-11-14 14:25:35.559660703 +0100
+++ gvfs-1.4.3/daemon/gvfsjobpull.c	2013-11-14 14:26:15.892740081 +0100
@@ -32,7 +32,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobpull.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobPull, g_vfs_job_pull, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobpush.c.orig gvfs-1.4.3/daemon/gvfsjobpush.c
--- gvfs-1.4.3/daemon/gvfsjobpush.c.orig	2013-11-14 14:25:35.559660703 +0100
+++ gvfs-1.4.3/daemon/gvfsjobpush.c	2013-11-14 14:26:15.892740081 +0100
@@ -32,7 +32,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobpush.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobPush, g_vfs_job_push, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobqueryattributes.c.orig gvfs-1.4.3/daemon/gvfsjobqueryattributes.c
--- gvfs-1.4.3/daemon/gvfsjobqueryattributes.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobqueryattributes.c	2013-11-14 14:26:15.892740081 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobmove.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 #include "gvfsjobqueryattributes.h"
 
diff -up gvfs-1.4.3/daemon/gvfsjobqueryfsinfo.c.orig gvfs-1.4.3/daemon/gvfsjobqueryfsinfo.c
--- gvfs-1.4.3/daemon/gvfsjobqueryfsinfo.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobqueryfsinfo.c	2013-11-14 14:26:15.893740082 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobqueryfsinfo.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobQueryFsInfo, g_vfs_job_query_fs_info, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobqueryinfo.c.orig gvfs-1.4.3/daemon/gvfsjobqueryinfo.c
--- gvfs-1.4.3/daemon/gvfsjobqueryinfo.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobqueryinfo.c	2013-11-14 14:26:15.893740082 +0100
@@ -32,7 +32,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobqueryinfo.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobQueryInfo, g_vfs_job_query_info, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobsetattribute.c.orig gvfs-1.4.3/daemon/gvfsjobsetattribute.c
--- gvfs-1.4.3/daemon/gvfsjobsetattribute.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobsetattribute.c	2013-11-14 14:26:15.893740082 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobmove.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 #include "gvfsjobsetattribute.h"
 
diff -up gvfs-1.4.3/daemon/gvfsjobsetdisplayname.c.orig gvfs-1.4.3/daemon/gvfsjobsetdisplayname.c
--- gvfs-1.4.3/daemon/gvfsjobsetdisplayname.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobsetdisplayname.c	2013-11-14 14:26:15.893740082 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobsetdisplayname.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobSetDisplayName, g_vfs_job_set_display_name, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobstartmountable.c.orig gvfs-1.4.3/daemon/gvfsjobstartmountable.c
--- gvfs-1.4.3/daemon/gvfsjobstartmountable.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobstartmountable.c	2013-11-14 14:26:15.893740082 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobstartmountable.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobStartMountable, g_vfs_job_start_mountable, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobstopmountable.c.orig gvfs-1.4.3/daemon/gvfsjobstopmountable.c
--- gvfs-1.4.3/daemon/gvfsjobstopmountable.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobstopmountable.c	2013-11-14 14:26:15.894740084 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobstopmountable.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobStopMountable, g_vfs_job_stop_mountable, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobtrash.c.orig gvfs-1.4.3/daemon/gvfsjobtrash.c
--- gvfs-1.4.3/daemon/gvfsjobtrash.c.orig	2013-11-14 14:25:35.560660705 +0100
+++ gvfs-1.4.3/daemon/gvfsjobtrash.c	2013-11-14 14:26:15.894740084 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobtrash.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobTrash, g_vfs_job_trash, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobunmount.c.orig gvfs-1.4.3/daemon/gvfsjobunmount.c
--- gvfs-1.4.3/daemon/gvfsjobunmount.c.orig	2013-11-14 14:25:35.561660707 +0100
+++ gvfs-1.4.3/daemon/gvfsjobunmount.c	2013-11-14 14:26:15.894740084 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobunmount.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonprotocol.h"
 
 G_DEFINE_TYPE (GVfsJobUnmount, g_vfs_job_unmount, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/gvfsjobunmountmountable.c.orig gvfs-1.4.3/daemon/gvfsjobunmountmountable.c
--- gvfs-1.4.3/daemon/gvfsjobunmountmountable.c.orig	2013-11-14 14:25:35.561660707 +0100
+++ gvfs-1.4.3/daemon/gvfsjobunmountmountable.c	2013-11-14 14:26:15.894740084 +0100
@@ -31,7 +31,7 @@
 #include <dbus/dbus.h>
 #include <glib/gi18n.h>
 #include "gvfsjobunmountmountable.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsdaemonutils.h"
 
 G_DEFINE_TYPE (GVfsJobUnmountMountable, g_vfs_job_unmount_mountable, G_VFS_TYPE_JOB_DBUS)
diff -up gvfs-1.4.3/daemon/mount.c.orig gvfs-1.4.3/daemon/mount.c
--- gvfs-1.4.3/daemon/mount.c.orig	2013-11-14 14:25:35.561660707 +0100
+++ gvfs-1.4.3/daemon/mount.c	2013-11-14 14:26:15.894740084 +0100
@@ -32,7 +32,7 @@
 #include "mount.h"
 #include "gmountoperationdbus.h"
 #include "gvfsdaemonprotocol.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include <glib.h>
 #include <gio/gio.h>
 
diff -up gvfs-1.4.3/metadata/meta-daemon.c.orig gvfs-1.4.3/metadata/meta-daemon.c
--- gvfs-1.4.3/metadata/meta-daemon.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/metadata/meta-daemon.c	2013-11-14 14:26:15.903740102 +0100
@@ -28,7 +28,7 @@
 #include <locale.h>
 #include <stdlib.h>
 #include <dbus/dbus.h>
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "metatree.h"
 #include "gvfsdaemonprotocol.h"
 
diff -up gvfs-1.4.3/metadata/meta-set.c.orig gvfs-1.4.3/metadata/meta-set.c
--- gvfs-1.4.3/metadata/meta-set.c.orig	2009-06-24 10:27:13.000000000 +0200
+++ gvfs-1.4.3/metadata/meta-set.c	2013-11-14 14:26:15.909740114 +0100
@@ -3,7 +3,7 @@
 #include <glib/gstdio.h>
 #include <dbus/dbus.h>
 #include "gvfsdaemonprotocol.h"
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 
 static gboolean unset = FALSE;
 static gboolean list = FALSE;
diff -up gvfs-1.4.3/monitor/gphoto2/hal-pool.c.orig gvfs-1.4.3/monitor/gphoto2/hal-pool.c
--- gvfs-1.4.3/monitor/gphoto2/hal-pool.c.orig	2009-04-20 12:41:37.000000000 +0200
+++ gvfs-1.4.3/monitor/gphoto2/hal-pool.c	2013-11-14 14:26:15.912740120 +0100
@@ -23,7 +23,7 @@
 #include <string.h>
 #include <sys/time.h>
 
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "hal-pool.h"
 #include "hal-marshal.h"
diff -up gvfs-1.4.3/monitor/hal/hal-pool.c.orig gvfs-1.4.3/monitor/hal/hal-pool.c
--- gvfs-1.4.3/monitor/hal/hal-pool.c.orig	2009-11-27 15:13:48.000000000 +0100
+++ gvfs-1.4.3/monitor/hal/hal-pool.c	2013-11-14 14:26:15.912740120 +0100
@@ -23,7 +23,7 @@
 #include <string.h>
 #include <sys/time.h>
 
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "hal-pool.h"
 #include "hal-marshal.h"
diff -up gvfs-1.4.3/monitor/proxy/gproxydrive.c.orig gvfs-1.4.3/monitor/proxy/gproxydrive.c
--- gvfs-1.4.3/monitor/proxy/gproxydrive.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/monitor/proxy/gproxydrive.c	2013-11-14 14:26:15.918740131 +0100
@@ -30,7 +30,7 @@
 #include <glib.h>
 #include <glib/gi18n-lib.h>
 
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "gproxyvolumemonitor.h"
 #include "gproxydrive.h"
diff -up gvfs-1.4.3/monitor/proxy/gproxymount.c.orig gvfs-1.4.3/monitor/proxy/gproxymount.c
--- gvfs-1.4.3/monitor/proxy/gproxymount.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/monitor/proxy/gproxymount.c	2013-11-14 14:26:15.919740133 +0100
@@ -31,7 +31,7 @@
 #include <glib/gi18n-lib.h>
 #include <gio/gio.h>
 
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "gproxyvolumemonitor.h"
 #include "gproxymount.h"
diff -up gvfs-1.4.3/monitor/proxy/gproxymountoperation.c.orig gvfs-1.4.3/monitor/proxy/gproxymountoperation.c
--- gvfs-1.4.3/monitor/proxy/gproxymountoperation.c.orig	2009-09-21 16:28:18.000000000 +0200
+++ gvfs-1.4.3/monitor/proxy/gproxymountoperation.c	2013-11-14 14:26:15.920740135 +0100
@@ -27,7 +27,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "gproxymountoperation.h"
 
diff -up gvfs-1.4.3/monitor/proxy/gproxyshadowmount.c.orig gvfs-1.4.3/monitor/proxy/gproxyshadowmount.c
--- gvfs-1.4.3/monitor/proxy/gproxyshadowmount.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/monitor/proxy/gproxyshadowmount.c	2013-11-14 14:26:15.920740135 +0100
@@ -31,7 +31,7 @@
 #include <glib/gi18n-lib.h>
 #include <gio/gio.h>
 
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "gproxyvolumemonitor.h"
 #include "gproxyshadowmount.h"
diff -up gvfs-1.4.3/monitor/proxy/gproxyvolume.c.orig gvfs-1.4.3/monitor/proxy/gproxyvolume.c
--- gvfs-1.4.3/monitor/proxy/gproxyvolume.c.orig	2009-12-14 09:31:02.000000000 +0100
+++ gvfs-1.4.3/monitor/proxy/gproxyvolume.c	2013-11-14 14:26:15.921740137 +0100
@@ -31,7 +31,7 @@
 #include <glib/gi18n-lib.h>
 #include <gio/gio.h>
 
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "gproxydrive.h"
 #include "gproxyvolume.h"
diff -up gvfs-1.4.3/monitor/proxy/gproxyvolumemonitor.c.orig gvfs-1.4.3/monitor/proxy/gproxyvolumemonitor.c
--- gvfs-1.4.3/monitor/proxy/gproxyvolumemonitor.c.orig	2009-07-14 13:07:59.000000000 +0200
+++ gvfs-1.4.3/monitor/proxy/gproxyvolumemonitor.c	2013-11-14 14:26:15.922740139 +0100
@@ -35,7 +35,7 @@
 #include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <gio/gio.h>
-#include <gdbusutils.h>
+#include <gvfsdbusutils.h>
 
 #include "gproxyvolumemonitor.h"
 #include "gproxymount.h"
diff -up gvfs-1.4.3/monitor/proxy/gvfsproxyvolumemonitordaemon.c.orig gvfs-1.4.3/monitor/proxy/gvfsproxyvolumemonitordaemon.c
--- gvfs-1.4.3/monitor/proxy/gvfsproxyvolumemonitordaemon.c.orig	2009-09-21 16:28:18.000000000 +0200
+++ gvfs-1.4.3/monitor/proxy/gvfsproxyvolumemonitordaemon.c	2013-11-14 14:26:15.924740143 +0100
@@ -30,7 +30,7 @@
 #include <stdlib.h>
 #include <glib/gprintf.h>
 
-#include "gdbusutils.h"
+#include "gvfsdbusutils.h"
 #include "gvfsproxyvolumemonitordaemon.h"
 
 /* ---------------------------------------------------------------------------------------------------- */
