From 60f0ed90b3fa4d70f7896096dd837dd2277a89ac Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Fri, 14 Sep 2012 16:16:37 +0200
Subject: [PATCH] gdu volume monitor: Don't unref possibly NULL object

---
 monitor/gdu/ggduvolumemonitor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/monitor/gdu/ggduvolumemonitor.c b/monitor/gdu/ggduvolumemonitor.c
index 707faea..4021eab 100644
--- a/monitor/gdu/ggduvolumemonitor.c
+++ b/monitor/gdu/ggduvolumemonitor.c
@@ -126,7 +126,7 @@ g_gdu_volume_monitor_finalize (GObject *object)
 
   g_object_unref (monitor->mount_monitor);
 
-  g_object_unref (monitor->pool);
+  g_clear_object (&monitor->pool);
 
   list_free (monitor->last_optical_disc_devices);
   list_free (monitor->last_mountpoints);
-- 
2.1.0

